### PR TITLE
Add TOML cheatcode support

### PR DESCRIPTION
**Pull request by zerosnacks**
~~Pending the merge of https://github.com/foundry-rs/foundry/pull/7317~~ merged!

Pulls in the latest changes from Foundry, including TOML cheatcode support. Implements the `stdToml` abstraction mirroring `stdJson`.

### CODE REVIEW

**Comment by mds1** [comment]
Make sure to also run the `scripts/vm.py` file in this repo to automatically update `Vm.sol` with the new cheats

**Comment by zerosnacks** [comment]
> Make sure to also run the `scripts/vm.py` file in this repo to automatically update `Vm.sol` with the new cheats

Thanks @mds1! I've ran the script. It also includes the missing cheatcodes requested in #519 and more.

This PR is now ready for review :)

File: src/StdToml.sol [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+
+    function readBytes(string memory toml, string memory key) internal pure returns (bytes memory) {
+        return vm.parseTomlBytes(toml, key);
+    }
+
+    function readBytesArray(string memory toml, string memory key) internal pure returns (bytes[] memory) {
+        return vm.parseTomlBytesArray(toml, key);
+    }
+
+    function serialize(string memory jsonKey, string memory rootObject) internal returns (string memory) {
```
Comment (mds1) [inline src/StdToml.sol:86]: Looks like a few `json` references slipped by, can you rename the rest of them?
```suggestion
    function serialize(string memory to'lKey, string memory rootObject) internal returns (string memory) {
```

File: src/StdToml.sol [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+// ```
+// using stdToml for string;
+// string memory toml = vm.readFile("some_path");
+// toml.parseUint("<json_path>");
+// ```
+// To write:
+// ```
+// using stdToml for string;
+// string memory toml = "[key]\n a = 1\n b = 2";
+// toml.serialize("key", "c", 3);
```
Comment (mds1) [inline src/StdToml.sol:19]: just curious why the example here is a bit different/shorter than the one in the StdJson.sol comments?

File: src/StdToml.sol (hunk 1 above)
Comment (mds1) [reply src/StdToml.sol:86]: Hm, but the example in the comments at the top show this operating on TOML:
```solidity
using stdToml for string;
string memory toml = "[key]\n a = 1\n b = 2";
toml.serialize("key", "c", 3);
toml.write("<some_path>");
```

Comment (zerosnacks) [reply src/StdToml.sol:88]: Hi @mds1 I've updated the example with the correct code. This code was an initial draft and incorrect.

File: src/StdToml.sol (hunk 2 above)
Comment (zerosnacks) [reply src/StdToml.sol]: Example code was incorrect, please see the updated example

**Review by mds1 — APPROVED** [review]
Thanks!

**Pull request merged**